packages/mobile-top-nav/src/search-menu.js

```diff
@@ -4,106 +4,99 @@ class SearchMenu extends LitElement {
   static get properties() {
     return {
       searchMenuOpen: { type: Boolean },
-      searchMenuAnimate: { type: Boolean }
+      searchMenuAnimate: { type: Boolean },
     };
   }
 
   render() {
-    const searchMenuClass = this.searchMenuOpen ? 'search-menu open slide-in' : this.searchMenuAnimate ? 'search-menu slide-out' : 'search-menu';
-    const searchMenuHidden = this.searchMenuOpen ? 'false' : 'true';
-    const searchMenuExpanded = this.searchMenuOpen ? 'true' : 'false';
+    let searchMenuClass = 'initial';
+    if (this.searchMenuOpen) {
+      searchMenuClass = 'open';
+    }
+    if (!this.searchMenuOpen && this.searchMenuAnimate) {
+      searchMenuClass = 'closed';
+    }
+
+    const searchMenuHidden = Boolean(!this.searchMenuOpen).toString();
+    const searchMenuExpanded = Boolean(this.searchMenuOpen).toString();
+
+    const searchTypes = [
+      { option: 'metadata', label: 'Metadata' },
+      { option: 'text', label: 'text contents' },
+      { option: 'tv', label: 'TV news captions' },
+      { option: 'web', label: 'archived websites' },
+    ].map(({ option, label }) => {
+      const checked = option === 'metadata' ? 'checked' : '';
+
+      return html`
+        <label class="search-type">
+          <input type="radio" name="search" value="${option}" checked="${checked}" />
+          <span>Search ${label}</span>
+        </label>
+      `;
+    });
```

Comment:
One thing I've done to help slim down the `render()` method is break it up into smaller template pieces and pull them in to the main `render()`, ie:

```
render() {
  return html`
    ${this.searchTypesTemplate}
  `;
}

get searchTypesTemplate() {
  const searchTypes = [
    { option: 'metadata', label: 'Metadata' },
    ...
  ].map(({ option, label }) => {
    const checked = option === 'metadata' ? 'checked' : '';

    return html`
      <label class="search-type">
         ...
      </label>
    `;
  });
  return searchTypes;
}
```

Reply:
love that syntax, no explicit `()` needed. thanks for the insight